tokio/src/net/tcp/stream.rs

```diff
@@ -664,6 +670,77 @@ impl TcpStream {
         self.io.set_nodelay(nodelay)
     }
 
+    /// Reads the linger duration for this socket by getting the `SO_LINGER`
+    /// option.
+    ///
+    /// For more information about this option, see [`set_linger`].
+    ///
+    /// [`set_linger`]: TcpStream::set_linger
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use tokio::net::TcpStream;
+    ///
+    /// # async fn dox() -> Result<(), Box<dyn std::error::Error>> {
+    /// let stream = TcpStream::connect("127.0.0.1:8080").await?;
+    ///
+    /// println!("{:?}", stream.linger()?);
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn linger(&self) -> io::Result<Option<Duration>> {
+        let mio_socket = self.to_mio();
+
+        let result = mio_socket.get_linger();
+
+        std::mem::forget(mio_socket);
```

Comment:
Could we change this to use https://doc.rust-lang.org/std/mem/struct.ManuallyDrop.html around `self.to_mio()`? That is a bit safer in case `get_linger()` panics (which it shouldn't, but it is in a different lib).